api-reference/10 UI Widgets/dxFileManager/5 File System Providers/FileSystemProviderBase/3 Methods/copyItems().md

```diff
@@ -3,16 +3,15 @@ id: FileSystemProviderBase.copyItems()
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Copies files or folders.
 
 ##### return: Array<Promise<any>>
-<!-- Description goes here -->
+An array of Promise objects that are resolved after the files or folders are copied. It is a <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise" target="_blank">native Promise</a> or a <a href="http://api.jquery.com/Types/#Promise" target="_blank">jQuery.Promise</a> when you use jQuery.
```

Comment:
It's unclear why there are several promises and when they get resolved. This applies to all methods that return an array of promises.

Reply:
the programmer's answer: each promise is for each copied file. A promise with N-index gets resolved when N-indexed item is processed (copied, in this case).

Reply:
This is unclear from the docs.

Reply:
fixed